docs/standard/datetime/timeprovider-overview.md

```diff
@@ -0,0 +1,87 @@
+---
+title: What is the TimeProvider class
+description: Learn about the TimeProvider class in .NET and .NET Framework. TimeProvider provides an abstraction over time.
+ms.date: 12/03/2024
+ms.topic: overview
+dev_langs:
+  - "csharp"
+  - "vb"
+helpviewer_keywords:
+  - "TimeProvider class"
+  - "date and time classes [.NET]"
+#customer intent: As a developer, I want to understand what TimeProvider is so that I can use it.
+---
+
+# What is TimeProvider?
+
+<xref:System.TimeProvider?displayProperty=fullName> is an abstraction of time that provides a point in time as a <xref:System.DateTimeOffset> type. By using `TimeProvider`, you ensure that your code is testable and predictable. `TimeProvider` was introduced in .NET 8 and is also available for .NET Framework 4.7+ and .NET Standard 2.0 as a NuGet package.
+
+The <xref:System.TimeProvider> class defines the following capabilities:
+
+- Provides access to the date and time through <xref:System.TimeProvider.GetUtcNow?displayProperty=nameWithType> and <xref:System.TimeProvider.GetLocalNow?displayProperty=nameWithType>.
+- High-frequency timestamps with <xref:System.TimeProvider.GetTimestamp?displayProperty=nameWithType>.
+- Measure time between two timestamps with <xref:System.TimeProvider.GetElapsedTime*?displayProperty=nameWithType>.
+- High-resolution timers with <xref:System.TimeProvider.CreateTimer(System.Threading.TimerCallback,System.Object,System.TimeSpan,System.TimeSpan)?displayProperty=nameWithType>.
+- Get the current timezone with <xref:System.TimeProvider.LocalTimeZone?displayProperty=nameWithType>.
+
+## Default implementation
+
+.NET provides an implementation of <xref:System.TimeProvider> through the <xref:System.TimeProvider.System?displayProperty=nameWithType> property, with the following characteristics:
+
+- Date and time are calculated with <xref:System.DateTimeOffset.UtcNow?displayProperty=nameWithType> and <xref:System.TimeZoneInfo.Local?displayProperty=nameWithType>.
+- Timestamps are provided by <xref:System.Diagnostics.Stopwatch?displayProperty=fullName>.
+- Timers are implemented through an internal class and exposed as <xref:System.Threading.ITimer?displayProperty=nameWithType>.
+
+The following example demonstrates using <xref:System.TimeProvider> to get the current date and time:
+
+:::code language="csharp" source="./snippets/timeprovider-overview/csharp/Program.cs" id="GetLocal":::
+:::code language="vb" source="./snippets/timeprovider-overview/vb/Program.vb" id="GetLocal":::
+
+The following example demonstrates capturing elapsed time with <xref:System.TimeProvider.GetTimestamp?displayProperty=nameWithType>:
+
+:::code language="csharp" source="./snippets/timeprovider-overview/csharp/Program.cs" id="Timestamp":::
+:::code language="vb" source="./snippets/timeprovider-overview/vb/Program.vb" id="Timestamp":::
+
+## FakeTimeProvider implementation
```

Comment:
>FakeTimeProvider implementation [](http://example.com/codeflow?start=3&length=31)

could you please add a link to this class docs? 

[https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.time.testing.faketimeprovider?view=net-9.0-pp](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.time.testing.faketimeprovider?view=net-9.0-pp)

Reply:
This link exists on line 49